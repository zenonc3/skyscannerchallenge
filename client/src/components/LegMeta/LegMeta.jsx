import PropTypes from 'prop-types';
import React from 'react';
import BpkText from 'bpk-component-text/src/BpkText';


const formatNumStops = (numStops) => {
  switch (numStops) {
    case 0:
      return 'Direct';
    case 1:
      return '1 stop';
    default:
      return `${numStops} stops`;
  }
};

const formatDuration = (durationMinutes) => {
  const hours = Math.floor(durationMinutes / 60);
  const minutes = durationMinutes % 60;
  return `${hours}h ${minutes}`;
};

const LegMeta = ({ duration, numStops }) => (
  <React.Fragment>
    <BpkText tagName="p" textStyle="sm">{formatDuration(duration)}</BpkText>
    <BpkText tagName="p" textStyle="sm">{formatNumStops(numStops)}</BpkText>
  </React.Fragment>
);

LegMeta.propTypes = {
  duration: PropTypes.number.isRequired,
  numStops: PropTypes.number.isRequired,
};

export default LegMeta;
