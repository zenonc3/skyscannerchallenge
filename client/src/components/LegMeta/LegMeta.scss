.LegMeta {
  &__stops {
    color: rgb(255, 123, 89);
  }
}
