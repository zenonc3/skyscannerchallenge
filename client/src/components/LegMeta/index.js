import LegMeta from './LegMeta';

export default LegMeta;
