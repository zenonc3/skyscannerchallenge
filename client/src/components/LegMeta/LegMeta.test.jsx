import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import LegMeta from './LegMeta';

describe('Leg Meta', () => {
  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<LegMeta duration={20} numStops={2} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

