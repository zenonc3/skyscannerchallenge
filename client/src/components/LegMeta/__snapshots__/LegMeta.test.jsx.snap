// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leg Meta should render correctly 1`] = `
<Unknown>
  <BpkText
    bold={false}
    className={null}
    tagName="p"
    textStyle="sm"
  >
    0h 20
  </BpkText>
  <BpkText
    bold={false}
    className={null}
    tagName="p"
    textStyle="sm"
  >
    2 stops
  </BpkText>
</Unknown>
`;
